### PR TITLE
contrib/gomodule/redigo: Don't pass nil contexts around in withSpan

### PR DESCRIPTION
**Pull request by abustany**
This fixes issue #710.

### CODE REVIEW

File: contrib/gomodule/redigo/redigo.go [hunk 1]
```diff
@@ -136,6 +136,15 @@ func withSpan(do func(commandName string, args ...interface{}) (interface{}, err
 		}
 	}
 
+	if ctx == nil {
+		// no context (or a nil context) was passed by the caller, but withSpan
+		// and the tracer functions rely on having a context.
+		//
```
Comment (gbbr) [inline contrib/gomodule/redigo/redigo.go:142]: I think we can remove this part of the comment.

File: contrib/gomodule/redigo/redigo.go [hunk 2]
```diff
@@ -136,6 +136,15 @@ func withSpan(do func(commandName string, args ...interface{}) (interface{}, err
...
+		// Passing a nil context was working up to Go 1.15, where issue #37908
+		// was closed.
+		ctx = context.Background()
+	}
+
```
Comment (gbbr) [inline contrib/gomodule/redigo/redigo.go:147]: Can we please remove this empty line?

**Review by gbbr — COMMENTED** [review]
Thanks for doing this

**Comment by abustany** [comment]
Thanks for the quick review! I've [addressed the comments](https://github.com/DataDog/dd-trace-go/compare/8376dd1e66e4fadc283f3064371222cb731ef734..eb4207e2ae9edea746be0d78da7248f3de502946).

**Review by gbbr — APPROVED** [review]
Thanks a lot!

**Comment by jontonsoup** [comment]
is there a schedule on when this will be cut into a release? it prevents us from upgrading to 15 currently :)

**Pull request merged**